rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -407,7 +421,20 @@ private void initStatus(WorkflowRun run) {
         } else if (execution.getCauseOfFailure() != null) {
             setStatus(StatusExt.valueOf(execution.getCauseOfFailure()));
         } else if (execution.isComplete()) {
-            setStatus(StatusExt.SUCCESS);
+            Result r = run.getResult();
+            if (r == Result.NOT_BUILT) {
```

Comment:
@jtnord What are you proposing?  In this case I'm doing a 1:1 mapping here (I'd do a mapping with switch/case if the types allowed for it).